SignalR Redis Configuration needs update

### Description

The existing .net 8 documentation for SignalR horizontal scaling with Redis tells us to configure it like this:
```
services.AddSignalR()
  .AddStackExchangeRedis(connectionString, options => {
      options.Configuration.ChannelPrefix = "MyApp";
  });
```

When we use the existing documentation to configure a redis backplane for SignalR, we receive a warning that says:
```
warning CS0618: 'RedisChannel.implicit operator RedisChannel(string)' is obsolete: 
'It is preferable to explicitly specify a PatternMode, or use the Literal/Pattern methods' 
```

Please update the .Net 8 documentation to have a signalr setup that does not produce warnings. 

### Page URL

https://learn.microsoft.com/en-us/aspnet/core/signalr/redis-backplane?view=aspnetcore-8.0

### Content source URL

https://github.com/dotnet/AspNetCore.Docs/blob/main/aspnetcore/signalr/redis-backplane.md

### Document ID

bddd7575-422d-4342-af48-f6b0979fb69a

### Article author

@bradygaster

--- a/aspnetcore/signalr/redis-backplane.md
+++ b/aspnetcore/signalr/redis-backplane.md
@@ -5,7 +5,7 @@ description: Learn how to set up a Redis backplane to enable scale-out for an AS
 monikerRange: '>= aspnetcore-2.1'
 ms.author: bradyg
 ms.custom: mvc
-ms.date: 11/12/2019
+ms.date: 02/06/2024
 uid: signalr/redis-backplane
 ---
 
@@ -34,10 +34,10 @@ This article explains SignalR-specific aspects of setting up a [Redis](https://r
 
   * `Microsoft.AspNetCore.SignalR.StackExchangeRedis`
   
-* In the `Startup.ConfigureServices` method, call <xref:Microsoft.Extensions.DependencyInjection.StackExchangeRedisDependencyInjectionExtensions.AddStackExchangeRedis*>:
+* Call <xref:Microsoft.Extensions.DependencyInjection.StackExchangeRedisDependencyInjectionExtensions.AddStackExchangeRedis*> by adding the following line before the line that calls `builder.Build()`) in `Program.cs` file.
 
   ```csharp
-  services.AddSignalR().AddStackExchangeRedis("<your_Redis_connection_string>");
+  builder.Services.AddSignalR().AddStackExchangeRedis("<your_Redis_connection_string>");
   ```
   
 * Configure options as needed:
@@ -47,9 +47,9 @@ This article explains SignalR-specific aspects of setting up a [Redis](https://r
   The following example shows how to set options in the `ConfigurationOptions` object. This example adds a channel prefix so that multiple apps can share the same Redis instance, as explained in the following step.
 
   ```csharp
-  services.AddSignalR()
+  builder.Services.AddSignalR()
     .AddStackExchangeRedis(connectionString, options => {
-        options.Configuration.ChannelPrefix = "MyApp";
+        options.Configuration.ChannelPrefix = RedisChannel.Literal("MyApp");
     });
   ```
 

--- a/aspnetcore/signalr/redis-backplane.md
+++ b/aspnetcore/signalr/redis-backplane.md
@@ -22,7 +22,7 @@ This article explains SignalR-specific aspects of setting up a [Redis](https://r
 * Deploy a Redis server.
 
   > [!IMPORTANT]
-  > For production use, a Redis backplane is recommended only when it runs in the same data center as the SignalR app. Otherwise, network latency degrades performance. If your SignalR app is running in the Azure cloud, we recommend Azure SignalR Service instead of a Redis backplane. 
+  > For production use, a Redis backplane is recommended only when it runs in the same data center as the SignalR app. Otherwise, network latency degrades performance. If your SignalR app is running in the Azure cloud, we recommend Azure SignalR Service instead of a Redis backplane.
 
   For more information, see the following resources:
 
@@ -34,7 +34,7 @@ This article explains SignalR-specific aspects of setting up a [Redis](https://r
 
   * `Microsoft.AspNetCore.SignalR.StackExchangeRedis`
   
-* Call <xref:Microsoft.Extensions.DependencyInjection.StackExchangeRedisDependencyInjectionExtensions.AddStackExchangeRedis*> by adding the following line before the line that calls `builder.Build()`) in `Program.cs` file.
+* Call <xref:Microsoft.Extensions.DependencyInjection.StackExchangeRedisDependencyInjectionExtensions.AddStackExchangeRedis*> by adding the following line before the line that calls `builder.Build()`) in the `Program.cs` file.
 
   ```csharp
   builder.Services.AddSignalR().AddStackExchangeRedis("<your_Redis_connection_string>");

--- a/aspnetcore/signalr/redis-backplane/includes/redis-backplane6.md
+++ b/aspnetcore/signalr/redis-backplane/includes/redis-backplane6.md
@@ -19,10 +19,10 @@ This article explains SignalR-specific aspects of setting up a [Redis](https://r
 
   * `Microsoft.AspNetCore.SignalR.StackExchangeRedis`
   
-* In the `Startup.ConfigureServices` method, call <xref:Microsoft.Extensions.DependencyInjection.StackExchangeRedisDependencyInjectionExtensions.AddStackExchangeRedis*>:
+* Call <xref:Microsoft.Extensions.DependencyInjection.StackExchangeRedisDependencyInjectionExtensions.AddStackExchangeRedis*> by adding the following line before the line that calls `builder.Build()`) in `Program.cs` file.
 
   ```csharp
-  services.AddSignalR().AddStackExchangeRedis("<your_Redis_connection_string>");
+  builder.Services.AddSignalR().AddStackExchangeRedis("<your_Redis_connection_string>");
   ```
   
 * Configure options as needed:
@@ -32,9 +32,9 @@ This article explains SignalR-specific aspects of setting up a [Redis](https://r
   The following example shows how to set options in the `ConfigurationOptions` object. This example adds a channel prefix so that multiple apps can share the same Redis instance, as explained in the following step.
 
   ```csharp
-  services.AddSignalR()
+  builder.Services.AddSignalR()
     .AddStackExchangeRedis(connectionString, options => {
-        options.Configuration.ChannelPrefix = "MyApp";
+        options.Configuration.ChannelPrefix = RedisChannel.Literal("MyApp");
     });
   ```
 

--- a/aspnetcore/signalr/redis-backplane/includes/redis-backplane6.md
+++ b/aspnetcore/signalr/redis-backplane/includes/redis-backplane6.md
@@ -7,7 +7,7 @@ This article explains SignalR-specific aspects of setting up a [Redis](https://r
 * Deploy a Redis server.
 
   > [!IMPORTANT]
-  > For production use, a Redis backplane is recommended only when it runs in the same data center as the SignalR app. Otherwise, network latency degrades performance. If your SignalR app is running in the Azure cloud, we recommend Azure SignalR Service instead of a Redis backplane. 
+  > For production use, a Redis backplane is recommended only when it runs in the same data center as the SignalR app. Otherwise, network latency degrades performance. If your SignalR app is running in the Azure cloud, we recommend Azure SignalR Service instead of a Redis backplane.
 
   For more information, see the following resources:
 
@@ -19,7 +19,7 @@ This article explains SignalR-specific aspects of setting up a [Redis](https://r
 
   * `Microsoft.AspNetCore.SignalR.StackExchangeRedis`
   
-* Call <xref:Microsoft.Extensions.DependencyInjection.StackExchangeRedisDependencyInjectionExtensions.AddStackExchangeRedis*> by adding the following line before the line that calls `builder.Build()`) in `Program.cs` file.
+* Call <xref:Microsoft.Extensions.DependencyInjection.StackExchangeRedisDependencyInjectionExtensions.AddStackExchangeRedis*> by adding the following line before the line that calls `builder.Build()`) in the `Program.cs` file.
 
   ```csharp
   builder.Services.AddSignalR().AddStackExchangeRedis("<your_Redis_connection_string>");

--- a/aspnetcore/signalr/redis-backplane/includes/redis-backplane7.md
+++ b/aspnetcore/signalr/redis-backplane/includes/redis-backplane7.md
@@ -19,10 +19,10 @@ This article explains SignalR-specific aspects of setting up a [Redis](https://r
 
   * `Microsoft.AspNetCore.SignalR.StackExchangeRedis`
   
-* In the `Startup.ConfigureServices` method, call <xref:Microsoft.Extensions.DependencyInjection.StackExchangeRedisDependencyInjectionExtensions.AddStackExchangeRedis*>:
+* Call <xref:Microsoft.Extensions.DependencyInjection.StackExchangeRedisDependencyInjectionExtensions.AddStackExchangeRedis*> by adding the following line before the line that calls `builder.Build()`) in `Program.cs` file.
 
   ```csharp
-  services.AddSignalR().AddStackExchangeRedis("<your_Redis_connection_string>");
+  builder.Services.AddSignalR().AddStackExchangeRedis("<your_Redis_connection_string>");
   ```
   
 * Configure options as needed:

--- a/aspnetcore/signalr/redis-backplane/includes/redis-backplane7.md
+++ b/aspnetcore/signalr/redis-backplane/includes/redis-backplane7.md
@@ -7,7 +7,7 @@ This article explains SignalR-specific aspects of setting up a [Redis](https://r
 * Deploy a Redis server.
 
   > [!IMPORTANT]
-  > For production use, a Redis backplane is recommended only when it runs in the same data center as the SignalR app. Otherwise, network latency degrades performance. If your SignalR app is running in the Azure cloud, we recommend Azure SignalR Service instead of a Redis backplane. 
+  > For production use, a Redis backplane is recommended only when it runs in the same data center as the SignalR app. Otherwise, network latency degrades performance. If your SignalR app is running in the Azure cloud, we recommend Azure SignalR Service instead of a Redis backplane.
 
   For more information, see the following resources:
 
@@ -19,7 +19,7 @@ This article explains SignalR-specific aspects of setting up a [Redis](https://r
 
   * `Microsoft.AspNetCore.SignalR.StackExchangeRedis`
   
-* Call <xref:Microsoft.Extensions.DependencyInjection.StackExchangeRedisDependencyInjectionExtensions.AddStackExchangeRedis*> by adding the following line before the line that calls `builder.Build()`) in `Program.cs` file.
+* Call <xref:Microsoft.Extensions.DependencyInjection.StackExchangeRedisDependencyInjectionExtensions.AddStackExchangeRedis*> by adding the following line before the line that calls `builder.Build()`) in the `Program.cs` file.
 
   ```csharp
   builder.Services.AddSignalR().AddStackExchangeRedis("<your_Redis_connection_string>");

